api/v1alpha3/types.go

```diff
@@ -311,6 +311,8 @@ type OSDisk struct {
 	DiskSizeGB       int32             `json:"diskSizeGB"`
 	ManagedDisk      ManagedDisk       `json:"managedDisk"`
 	DiffDiskSettings *DiffDiskSettings `json:"diffDiskSettings,omitempty"`
+	// +optional
+	CachingType string `json:"cachingType"`
```

Comment:
I suspect this should be `` `json:"cachingType,omitempty"` ``? Need to test

Reply:
In this case wouldn't omitempty only have an effect between the api request and mutating webhook (since we're defaulting that field)?  Also for my understanding-- are you referring to the effect of serializing an empty field on the virtualmachine service (if an empty field exists it might cause an error as opposed to a nil field)?  Thanks for the help! :smile:

Reply:
you're correct about behavior with the webhook, I think my memory is picking up on an old controller-gen bug where `+optional` wasn't respected so `,omitempty` was actually required for the correct behavior.